File: Day 25 : Running Time and Complexity/Solution.py
from math import sqrt

def notPrime(num: int) -> bool:
    for i in range(2, int(sqrt(num))+1):
        if num % i == 0:
            return True
    return False

if __name__ == "__main__":
    N = int(input())
    for i in range(N):
        num = int(input())
        if num <= 1:
            print("Not prime")
        elif(num > 1 and notPrime(num)):
            print("Not prime")
        else:
            print("Prime")

File: Day 25 : Running Time and Complexity/Solution.java
import java.util.*;

public class Solution {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int T = scan.nextInt();
        for (int j = 0; j < T; j++) {
            int num = scan.nextInt();
            if (num <= 1) {
                System.out.println("Not prime");
            } else {
                int divisors = 1;
                for (int i = 2; i <= Math.sqrt(num); i++) {
                    if (num % i == 0) {
                        System.out.println("Not prime");
                        divisors++;
                        break;
                    }
                }
                if (divisors == 1)
                    System.out.println("Prime");
            }
        }
        scan.close();
    }
}
